components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java

```diff
@@ -221,6 +231,137 @@ public void deleteBrandingPreference(String type, String name, String locale)
+            }
+            return buildCustomTextFromResource(inputStream, type, name, screen, locale);
+        } catch (ConfigurationManagementException e) {
+            if (RESOURCE_NOT_EXISTS_ERROR_CODE.equals(e.getErrorCode())) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Can not find a custom text preference configurations for tenant: " + tenantDomain, e);
+                }
+                throw handleClientException(ERROR_CODE_CUSTOM_TEXT_PREFERENCE_NOT_EXISTS, tenantDomain);
+            }
+            throw handleServerException(ERROR_CODE_ERROR_GETTING_CUSTOM_TEXT_PREFERENCE, tenantDomain, e);
+        } catch (IOException e) {
+            throw handleServerException(ERROR_CODE_ERROR_BUILDING_CUSTOM_TEXT_PREFERENCE, tenantDomain);
+        }
+    }
+
+    public CustomText resolveCustomText(String type, String name, String screen, String locale)
+            throws BrandingPreferenceMgtException {
+
+        if (BrandingPreferenceManagerComponentDataHolder.getInstance().getUiBrandingPreferenceResolver() != null) {
+            return BrandingPreferenceManagerComponentDataHolder.getInstance().getUiBrandingPreferenceResolver()
+                    .resolveCustomText(type, name, screen, locale);
+        }
+        return getCustomText(type, name, screen, locale);
+    }
+
+    public CustomText replaceCustomText(CustomText customText)
+            throws BrandingPreferenceMgtException {
+
+        String resourceName = getResourceNameForCustomText(customText.getScreen(), customText.getLocale());
+        String tenantDomain = getTenantDomain();
+        // Check whether the branding resource exists in the particular tenant.
+        if (!isResourceExists(CUSTOM_TEXT_RESOURCE_TYPE, resourceName)) {
+            throw handleClientException(ERROR_CODE_CUSTOM_TEXT_PREFERENCE_NOT_EXISTS, tenantDomain);
+        }
+
+        String preferencesJSON = generatePreferencesJSONFromPreference(customText.getPreference());
+        if (!BrandingPreferenceMgtUtils.isValidJSONString(preferencesJSON)) {
+            throw handleClientException(ERROR_CODE_INVALID_BRANDING_PREFERENCE, tenantDomain);
+        }
+
+        try {
+            InputStream inputStream = new ByteArrayInputStream(preferencesJSON.getBytes(StandardCharsets.UTF_8));
+            Resource customTextResource = buildResource(resourceName, inputStream);
+            getConfigurationManager().replaceResource(CUSTOM_TEXT_RESOURCE_TYPE, customTextResource);
+        } catch (ConfigurationManagementException e) {
+            throw handleServerException(ERROR_CODE_ERROR_UPDATING_CUSTOM_TEXT_PREFERENCE, tenantDomain, e);
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Custom Text preference for tenant: " + tenantDomain + " replaced successfully.");
+        }
+        return customText;
+    }
+
+    public void deleteCustomText(String type, String name, String screen, String locale)
```

Comment:
add "@override" annotation for these public methods, else need method comments to public methods

Reply:
Adddressed with https://github.com/wso2-extensions/identity-branding-preference-management/commit/fdd3486cfdc51558c723795cf2fb75b6faafcdc5